ShopAppTests/App/Account/ViewModels/BaseAddressListViewModelSpec.swift

```diff
@@ -0,0 +1,201 @@
+//
+//  BaseAddressListViewModelSpec.swift
+//  ShopAppTests
+//
+//  Created by Evgeniy Antonov on 3/1/18.
+//  Copyright © 2018 RubyGarage. All rights reserved.
+//
+
+import Nimble
+import Quick
+import RxSwift
+import ShopApp_Gateway
+
+@testable import ShopApp
+
+class BaseAddressListViewModelSpec: QuickSpec {
+    override func spec() {
+        var viewModel: BaseAddressListViewModel!
+        var customerUseCaseMock: CustomerUseCaseMock!
+        var deleteAddressUseCaseMock: DeleteAddressUseCaseMock!
+        var updateDefaultAddressUseCaseMock: UpdateDefaultAddressUseCaseMock!
+        
+        beforeEach {
+            let authenticationRepositoryMock = AuthentificationRepositoryMock()
+            customerUseCaseMock = CustomerUseCaseMock(repository: authenticationRepositoryMock)
+            let paymentsRepositoryMock = PaymentsRepositoryMock()
+            updateDefaultAddressUseCaseMock = UpdateDefaultAddressUseCaseMock(repository: paymentsRepositoryMock)
+            deleteAddressUseCaseMock = DeleteAddressUseCaseMock(repository: paymentsRepositoryMock)
+            viewModel = BaseAddressListViewModel(customerUseCase: customerUseCaseMock, updateDefaultAddressUseCase: updateDefaultAddressUseCaseMock, deleteAddressUseCase: deleteAddressUseCaseMock)
+        }
+        
+        describe("when view model initialized") {
```

Comment:
You also need to check initial values of `customerAddresses` and `customerDefaultAddress` variables.